lib/check.nix

```diff
@@ -1,7 +1,8 @@
-{ stdenv, lib, haskellLib, srcOnly }:
-drv:
+{ stdenv, lib, haskellLib, srcOnly, withCoverage ? false }:
+d:
```

Comment:
This seems a little bit circular.
How about removing the `withCoverage` arg and just copying all the `**/*.tix` files which exist - if any.
Then we don't need the `checkWithCoverage` variant.